src/components/common/store/store.tsx

```diff
@@ -31,6 +33,8 @@ const searchQueryResult = (state: RootState) =>
 
 const getComponentState = (state: RootState) => state.paramReducer;
 
-export { searchQueryResult, getComponentState };
+const getSearchbarExpanded = (state: RootState) => state.searchbar.isExpanded;
```

Comment:
Is it necessary to use store to remember state of the component? Can we determine it by text inside and then use similar approach as the map expand to signal the menu to collapse?

You can find code in map.tsx, that is you try to observer a component resize and act accordingly


      // Create a resize observer to the canvas so we know if its size have changed
      // and we need to redraw the map
      const resizeObserver = new ResizeObserver((entries) =>
        // https://stackoverflow.com/questions/70533564/mapbox-gl-flickers-when-resizing-the-container-div
        setTimeout(() => map.resize(), 0.1)
      );
      resizeObserver.observe(map.getContainer());

Reply:
Seems I made things complicated...I removed the change in redux and mange the search bar width in Header. Thanks!